src/resolver/descriptor.cc

```diff
@@ -457,6 +457,81 @@ to_json( nlohmann::json & jto, const ManifestDescriptorRaw & descriptor )
     }
 }
 
+/* -------------------------------------------------------------------------- */
+
+ManifestDescriptorRaw::ManifestDescriptorRaw(
+  const std::string_view & descriptor )
+{
+  size_t cursor = 0;
+  // Grab the input if it exists
+  if ( auto inputSepIdx
+       = descriptor.find_first_of( ManifestDescriptorRaw::inputSigil );
+       inputSepIdx != std::string_view::npos )
+    {
+      this->packageRepository
+        = std::string( descriptor.substr( cursor, inputSepIdx - cursor ) );
+      cursor = inputSepIdx + 1;
+    }
+  // Grab the attribute path or package name
+  size_t attrsEndIdx;
+  bool   hasVersion = false;
+  if ( auto versionSepIdx
+       = descriptor.find_first_of( ManifestDescriptorRaw::versionSigil );
```

Comment:
question nonblocking: why use `find_first_of`? Isn't that for checking multiple characters, not one like you are?

Reply:
Good question, I think I misread this as "find me the first occurrence of ...". It sounds like `find` already does that. Quick fix incoming.